zebrad/src/commands/start.rs

```diff
@@ -308,6 +309,31 @@ impl StartCmd {
         let scan_task_handle: tokio::task::JoinHandle<Result<(), Report>> =
             tokio::spawn(std::future::pending().in_current_span());
 
+        // And finally, spawn the internal Zcash miner, if it is enabled.
+        //
+        // TODO: add a config to enable the miner rather than a feature.
+        #[cfg(feature = "internal-miner")]
+        let miner_task_handle = {
+            info!("spawning Zcash miner");
+            let rpc = zebra_rpc::methods::get_block_template_rpcs::GetBlockTemplateRpcImpl::new(
+                config.network.network,
+                config.mining.clone(),
+                mempool,
+                read_only_state_service,
+                latest_chain_tip,
+                block_verifier_router,
+                sync_status,
+                address_book,
+            );
+
+            crate::components::miner::spawn_init(&config.mining, rpc)
+        };
+
+        #[cfg(not(feature = "internal-miner"))]
+        // Spawn a dummy miner task which doesn't do anything and never finishes.
+        let miner_task_handle: tokio::task::JoinHandle<Result<(), Report>> =
+            tokio::spawn(std::future::pending().in_current_span());
```

Comment:
Just curious, why use pending futures rather than the original miner/scan tasks?

Reply:
When the feature isn't enabled, the functions that create the miner/scan tasks aren't compiled, so we can't call them.